src/rp2_common/pico_clib_interface/picolibc_interface.c

```diff
@@ -39,6 +40,7 @@ static int picolibc_getc(__unused FILE *file) {
 #if LIB_PICO_STDIO
     return stdio_getchar();
 #endif
+    return -1;
```

Comment:
I guess this _could_ have been
```c
#if LIB_PICO_STDIO
    return stdio_getchar();
#else
    return -1;
#endif
```
but presumably most compilers will make that optimisation automatically?